src/chainstate/stacks/boot/bns.clar

```diff
@@ -878,6 +878,11 @@
           (> block-height (+ lifetime lease-started-at)) 
           (<= block-height (+ (+ lifetime lease-started-at) NAME_GRACE_PERIOD_DURATION)))))))
 
+(define-read-only (get-name (owner principal))
+  (match (map-get? owner-name owner)
+    name (ok name)
```

Comment:
Thank you @friedger. I think the current approach is incomplete, you'd want to make sure that this name is still resolving.
So you can pull from that map, use the method `name-resolve`, and then return a result if it returns `ok`.

Reply:
@lgalabru for that I would add a function get-resolved-name to keep flexibility for contract developers

Reply:
Then docs should be @agraebe :
("get-name", "Returns a response with the username that is associated with the user if any. The name can be resolvable, expired, revoked or in grace period. It is the responsibility of the function caller to act accordingly. If no name is associated with the user an error 2013 (`ERR_NAME_NOT_FOUND`) is returned.")

Reply:
Thank you for adding the tests!
I like this feature. Can we rename to `resolve-principal` or equivalent, so we increase consistency with `resolve-name`?
I still have some reserves concerning the `get-name`. If we can do an analogy with DNS, it's basically equivalent to resolving an expired domain.
Can you unpack some use cases? 
Thanks!

Reply:
@lgalabru I have renamed `get-resolvable-name` as suggested.

As use cases, a contract could act if a name is in grace period and maybe renew it. There are probably smart people out there that will find more use cases.

As the data is available through the RCP api anywhere, adding `get-name` unlocks this api to smart contracts until we have https://github.com/clarity-lang/reference/issues/30

Reply:
Yeah I'm not saying that this data is not available, I'm just saying that the contract should be in charge of exposing relevant data, and the `get-name` method looks like a foot-gun for developers.

If the name is in grace period, or expired, great, with your current implementation of `resolve-principal`, you will get the right error code.

Reply:
what you could do in resolve-principal, is, in case of an error, merge the tuple returned by resolve-name with the name that was queried.

Reply:
I like that.

I have remove `get-name` and added the name in the error. 
Thank you for your input.

Reply:
Yay, thank you. We don't have a CI for this tests, running on my machine and will follow-up.

Reply:
@friedger This PR, by updating BNS, is modifying the signature of the Genesis block. 
We have some testing around that, can you update the tests `test_chainstate_sampled_genesis_consistency` and `test_chainstate_full_genesis_consistency`?
Thank you!

Reply:
You should also merge (or rebase) the latest `next` into this branch, so that the hash in `test_chainstate_sampled_genesis_consistency` is up to date when we merge it into `next`.

Reply:
merged and updated!